mesonbuild/dependencies/detect.py

```diff
@@ -40,10 +40,14 @@ def __contains__(self, key: object) -> bool:
 
 def get_dep_identifier(name: str, kwargs: T.Dict[str, T.Any]) -> 'TV_DepID':
     identifier: 'TV_DepID' = (('name', name), )
+    from ..interpreter.type_checking import DEPENDENCY_KWS
+    nkwargs = {k.name: k.default for k in DEPENDENCY_KWS}
+    nkwargs.update(kwargs)
```

Comment:
So this spares a mismatch when... the default is manually specified? Or?

Can you elaborate on what problem this is trying to solve.

Reply:
Okay, basically what's happening is that `find_external_dependency()` leaves the kwargs dict with uninitialized values, while `func_dependency()` initializes all unset values to their defaults through `typed_kwargs`. This results in functionally equivalent calls to get a dependency having different cache keys

Reply:
Can you mention `find_external_dependency()` in the commit message? Otherwise this PR lgtm.